addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2331,6 +2346,50 @@ export class OdooEditor extends EventTarget {
                 }
             }
         } else if (node.nodeType !== Node.TEXT_NODE) {
+            if (node.nodeName === 'TD' && node.hasAttribute('style')) {
+                // TD tags do not support style in v15, move style to children or new SPAN.
+                const spans = node.childNodes.length === 0 || [...node.childNodes].some(n => !['SPAN', 'FONT'].includes(n.nodeName))
+                    ? [this.document.createElement('SPAN')]
+                    : [...node.childNodes];
```

Comment:
I'm confused by this line... If there are some `span` or `font` nodes then you will get _all_ the child nodes in the `spans` variable, regardless of whether those actually are spans (or font) or not. Isn't that weird ?

Reply:
No, I think you missed a `!`.
If there are no children or if some of the children are not styleable (are not spans or fonts), then create our own styleable wrapper.

Reply:
I like your first suggestion. It ensures we use SPAN and not FONT. It more correctly preserves the style inheritance mechanism (if a child node is later added).
I dislike the second suggestion because I prefer to keep the copy-paste operation idempotent.
What do you mean there
> only in 15.0 to boot

Reply:
I mean "in addition, it only applies to 15.0". Let's go for the first suggestion.

Reply:
Implemented the first suggestion. Works fine, but the case where we re-use node.firstChild is never met in the tests (there's always more than one child when children are span). Should I make sure we test that case ?

Reply:
Yes please 👍